js/bccm-customizations.js

```diff
@@ -2,49 +2,56 @@ import config from '/sitemedia/js/bccm-config.js';
 
 console.log(`bccm-customizations.js executing`) 
 
-
 // var FilterdUserGroups = defaultconfig.ruleButtonPlugin.userGroups.filter(filterUserGroups);
 
 var FilterdUserGroups = config.ruleButtonPlugin.userGroups.filter(filterUserGroups);
 
+var metadata_obj;
+
+
+fetch('/API/v2/items/' + manualRulePluginContext.item_id + '/metadata/').then(res => res.json()).then(data => metadata_obj = data);
+
+// maetadatagroup = metadata_obj.group_name
 
 //generates Button and adds EventListener
 
 function addManualRuleButton() {
     for (var i = 0; i < FilterdUserGroups.length; i++) {
         for (var x = 0; x < FilterdUserGroups[i].buttons.length; x++) {
-            let buttonConfig = FilterdUserGroups[i].buttons[x];
-            let ManualRuleButton = document.createElement("button");
-            ManualRuleButton.innerHTML = buttonConfig.label;
-            ManualRuleButton.type = "button";
-            ManualRuleButton.setAttribute("class", buttonConfig.button_colour);
-            document.getElementById("ManualRuleButtonDIV").appendChild(ManualRuleButton); 
-
-            ManualRuleButton.onclick = function() {
-                this.innerHTML = "working..."
-                this.disabled = true;
-                let THIS = this;
-                setTimeout(function(){THIS.disabled = false;THIS.innerHTML = buttonConfig.label;}, 5000);
-            }
+            if(FilterdUserGroups[i].buttons[x].metadata == metadata_obj.group_name){
```

Comment:
You should invert the if. For example:
```suggestion
            if(FilterdUserGroups[i].buttons[x].metadata != metadata_obj.group_name){
                continue;
            }
```

Also you should show the button on any type if the filter is not defined.